a5 onc 2015

bradymiller opened the pull request:
a5 item for onc 2015

continued from https://github.com/openemr/openemr/pull/3266

completed:
Added a new datatype that allows a comment to be included with list
Added a Unknown birth sex
Change the ethnicity and race datatype to use multselect list rather than single select list

sunsetsystems commented:
Reminds me of an idea for a SOC project: Move Demographics and History data items to vertical tables, as is done with encounter LBFs.

sjpadgett commented on interface/super/edit_layout.php:
```diff
@@ -84,6 +84,7 @@ function collectLayoutNames($condition, $mapping = '')
   "43" => xl("List box w/search"),
   "44" => xl("Multi-Select Facilties"),
   "45" => xl("Multi-Select Provider"),
+  "46" => xl("List box w/comment"),
```
I'm guessing nothing special needs to be handled in LBF/new.php

sunsetsystems commented:
Sorry to notice this so late... did you know that conditions in layout items can be used to hide or show a field depending on the value of another field? That might be a way to do what you want instead of creating a new data type.

sjpadgett commented:
which is very useful for questionnaires. now we just need to sum items:)

sunsetsystems commented:
> which is very useful for questionnaires. now we just need to sum items:)

LBF plug-ins can be used for more esoteric things like that. :)

bradymiller commented:
@sunsetsystems , thanks for the pointer. i'll check out the condition stuff to see if can avoid the new datatype.

bradymiller commented:
This PR will now officially pass ONC2015 item a5 :)
Here's what this PR does:
1. Added a new layout datatype that allows a comment to be included with list
2. Used the new datatype to support Gender Identity and Sexual Orientation selectors in demographics
3. Added a Unknown birth sex option
4. Changed the ethnicity and race datatype to use multselect list rather than single select list

Still some more road testing to do.

@sunsetsystems , did also try the Gender Identity and Sexual Orientation comment support via layout conditionals, which I could get to work. In this case, though, I like the idea of having the list determine which element(s) get a comment rather than relying on the layout entry that uses the list. This also ensure that it won't get broken by users that modify layouts since it is a core piece of ONC 2015. And now the user has 2 potential mechanisms to support this kind of feature. In the clinical realm (ie. LBF stuff), it is really common to have selectors to choose from where some of the options require a comment, so having this new datatype will be another good option in those scenarios. Hopefully that convinces you to embrace the new datatype :)

sunsetsystems commented:
Brady, consider it embraced. Just wanted to make sure you know about conditionals. Thanks for the improvements!

bradymiller commented:
testing well and bringing in

stephenwaite commented:
![excited_elephant](https://user-images.githubusercontent.com/5618372/110205722-bfc3f380-7e47-11eb-9190-4448245d932f.gif)

sjpadgett commented:
Puppies are cute, kittens are cute... baby elephants, not so much.:)
Check out bird in background saying to itself: "yeah, i'm remember when I was young once!"

stephenwaite commented:
had a lot of questions about this functionality for a few years, thanks again for the improvements @bradymiller !
